app/src/main/java/com/example/memeapp/data/service/implementation/MemesServiceImpl.kt

```diff
@@ -25,7 +26,26 @@ class MemesServiceImpl : MemesService {
         }
     }
 
+    override fun getSingleMeme(id: Int): Observable<MemesEntity> {
+        return Observable.create { subscriber ->
+            val callResponse = api.createService(MemeApi::class.java).getSingleMeme(id)
+            val response = callResponse.execute()
+
+            if (response.isSuccessful) {
+                val validResponse = response.body()?.data
+                validResponse?.let {
+                    subscriber.onNext(MemesMapper.transformMemes(it))
+                    subscriber.onComplete()
+                }
+            } else {
+                subscriber.onError(Throwable(response.message()))
+            }
+        }
+    }
+
     companion object {
         private const val CHOSEN_PAGE = 1
+        private const val ERROR_MESSAGE_EMPTY_RESPONSE =
+            "We can not get an answer because of an empty response"
```

Comment:
you will show this message on the screen? if so, this should have on the string resources

Reply:
Done thx